Exwm replace support

medranocalvo opened the pull request:
XELB changes needed for `replace-support` EXWM branch.

Names are not great, please suggest better if possible. It might be useful to rename or alias `xcb:+event` to `xcb:+add-event` for similarity with `add-hook` and `remove-hook`.

ch11ng commented on xcb.el:
```diff
@@ -39,6 +39,7 @@
 ;;   - `xcb:+request-unchecked+reply'
 ;; + Event handling
 ;;   - `xcb:+event'
+;;   - `xcb:+remove-event'
```
Use `xcb:remove-event`.

ch11ng commented on xcb.el:
```diff
@@ -458,6 +459,17 @@ classes of EVENT (since they have the same event number)."
+(cl-defmethod xcb:+remove-event ((obj xcb:connection) event listener)
+  "remove EVENT LISTENER
+
+Note that event listeners attached this way are shared with the super- and sub-
```
This docstring is no longer relevant.

ch11ng commented on xcb.el:
```diff
@@ -512,6 +524,10 @@ classes of EVENT (since they have the same event number)."
       (xcb:flush obj))))
 
+(cl-defmethod xcb:got-extension-data-p ((obj xcb:connection) namespace)
+  "Return t when the connection OBJ has extension data for NAMESPACE."
```
No need to convert the result to `t`.

ch11ng commented:
> It might be useful to rename or alias `xcb:+event` to `xcb:+add-event`.

The `+` in `xcb:+event` actually means 'add'.  `xcb:add-event` would be better.

medranocalvo committed:
Support WM_Sn atoms

* xcb-icccm.el (xcb:icccm:-atom, xcb:icccm:init): Register WM_Sn
atoms for each screen.

medranocalvo commented:
Thanks for the review.  Addressed the issues you noted.

Additionally, renamed `xcb:+event` to `xcb:add-event` and added the former as obsolete alias; let me know whether that's all right.

ch11ng commented:
Sorry but now that there is no use for `xcb:remove-event` (see the review on the other PR) perhaps we can just drop the changes on `xcb:+event` (the naming is admittedly not good but it's get used too much) and `xcb:remove-event`? Everything else looks fine to me.

PS.  I have no idea but some of my original reviews seem get lost.  This implementation of `xcb:remove-event` can't handle XGE and XKB events.

medranocalvo commented:
I didn't receive your comments w.r.t. XGE and XKB; you were right about it.
I think that being able to stop listening to events would be useful for XELB applications other than EXWM. Note that the old changes preserved the `xcb:+event` function as an alias.
I have removed those changes, along with `xcb:got-extension-data-p` which is not needed for replace support either.
